pennylane/ops/functions/is_commuting.py

```diff
@@ -374,165 +324,55 @@ def is_commuting(operation1, operation2):
-    # pylint: disable=too-many-branches
-    # pylint: disable=too-many-return-statements
-
-    if control_base.get(operation1.name) != "ControlledOperation":
-        control_base_1 = control_base.get(operation1.name)
-    else:
-        control_base_1 = operation1.control_base
-
-    if control_base.get(operation2.name) != "ControlledOperation":
-        control_base_2 = control_base.get(operation2.name)
-    else:
-        control_base_2 = operation2.control_base
-
-    target_wires_1 = qml.wires.Wires(
-        [w for w in operation1.wires if w not in operation1.control_wires]
-    )
-    target_wires_2 = qml.wires.Wires(
-        [w for w in operation2.wires if w not in operation2.control_wires]
-    )
-
-    control_control = intersection(operation1.control_wires, operation2.control_wires)
-    target_target = intersection(target_wires_1, target_wires_2)
-    control_target = intersection(operation1.control_wires, target_wires_2)
-    target_control = intersection(target_wires_1, operation2.control_wires)
-
-    # Case 2.1: disjoint targets
-    if control_control and not target_target and not control_target and not target_control:
-        return True
-
-    # Case 2.2: disjoint controls
-    if not control_control and target_target and not control_target and not target_control:
-        return _commutes(control_base_2, control_base_1)
+    op_set = {"U2", "U3", "Rot", "CRot"}
+    if operation1.name in op_set and operation2.name in op_set:
+        return check_commutation_two_non_simplified_rotations(operation1, operation2)
 
-    # Case 2.3: targets overlap and controls overlap
-    if target_target and control_control and not control_target and not target_control:
-        return _commutes(control_base_1, control_base_2)
+    control_base_1 = _get_target_name(operation1)
+    control_base_2 = _get_target_name(operation2)
 
-    # Case 2.4: targets and controls overlap
-    if control_target and target_control and not target_target:
-        return _commutes("ctrl", control_base_2) and _commutes("ctrl", control_base_1)
+    op1_control_wires = getattr(operation1, "control_wires", {})
+    op2_control_wires = getattr(operation2, "control_wires", {})
 
-    # Case 2.5: targets overlap with and controls and targets
-    if control_target and not target_control and target_target:
-        return _commutes("ctrl", control_base_2) and _commutes(control_base_1, control_base_2)
+    target_wires_1 = qml.wires.Wires([w for w in operation1.wires if w not in op1_control_wires])
+    target_wires_2 = qml.wires.Wires([w for w in operation2.wires if w not in op2_control_wires])
 
-    # Case 2.6: targets overlap with and controls and targets
-    if target_control and not control_target and target_target:
-        return _commutes(control_base_1, "ctrl") and _commutes(control_base_1, control_base_2)
+    ops_commute = True
+    if intersection(target_wires_1, target_wires_2):
+        ops_commute &= _commutes(control_base_1, control_base_2)
```

Comment:
Could we somehow avoid checking `_commutes` again if `ops_commute` is `False`?

Reply:
Even better! Excellent idea.